pyfr/__main__.py

```diff
@@ -176,23 +176,22 @@ def process_import(args):
 
 
 def process_partition_list(args):
-    with h5py.File(args.mesh, 'r') as mesh:
-        print('name', 'parts', sep=args.sep)
+    print('name', 'parts', sep=args.sep)
 
+    with h5py.File(args.mesh, 'r') as mesh:
         for name, part in sorted(mesh['partitionings'].items()):
             nparts = len(part['eles'].attrs['regions'])
             print(name, nparts, sep=args.sep)
 
 
 def process_partition_info(args):
     # Open the mesh
-    mesh = h5py.File(args.mesh, 'r')
-
-    # Read the partition region info from the mesh
-    regions = mesh[f'partitionings/{args.name}/eles'].attrs['regions']
+    with h5py.File(args.mesh, 'r') as mesh:
+        # Read the partition region info from the mesh
+        regions = mesh[f'partitionings/{args.name}/eles'].attrs['regions']
 
-    # Print out the header
-    print('part', *mesh['eles'], sep=args.sep)
+        # Print out the header
+        print('part', *mesh['eles'], sep=args.sep)
 
     # Compute and output the number of elements in each partition
     for i, neles in enumerate(regions[:, 1:] - regions[:, :-1]):
```

Comment:
`regions` comes from h5py so should be in the `with` block (h5py often returns proxy objects --- although not currently the case for attributes this might change in the future).